src/main/java/uk/gov/companieshouse/api/accounts/service/impl/CurrentPeriodServiceImpl.java

```diff
@@ -1,33 +1,165 @@
+        } catch (MongoException me) {
+            DataException dataException = new DataException(
+                    "Failed to insert " + ResourceName.SMALL_FULL.getName(), me);
+            LOGGER.errorContext(requestId, dataException, debugMap);
+            throw dataException;
+        }
+
+        smallFullService.addLink(companyAccountId, LinkType.SMALL_FULL, selfLink, requestId);
+
+        return new ResponseObject<>(ResponseStatus.CREATED, currentPeriod);
     }
 
     @Override
-    public String getResourceName() {
-        return "current-period";
+    public ResponseObject<CurrentPeriod> findById(String id, String requestId)
+            throws DataException {
+        CurrentPeriodEntity currentPeriodEntity;
+        try {
+            currentPeriodEntity = currentPeriodRepository.findById(id).orElse(null);
+        } catch (MongoException me) {
+            final Map<String, Object> debugMap = new HashMap<>();
+            debugMap.put("id", id);
+            DataException dataException = new DataException("Failed to find Current period", me);
+            LOGGER.errorContext(requestId, dataException, debugMap);
+            throw dataException;
+        }
+
+        if (currentPeriodEntity == null) {
+            return new ResponseObject<>(ResponseStatus.NOT_FOUND);
+        }
+        CurrentPeriod currentPeriod = currentPeriodTransformer.transform(currentPeriodEntity);
+        return new ResponseObject<>(ResponseStatus.FOUND, currentPeriod);
+    }
+
+    @Override
+    public void addLink(String id, LinkType linkType, String link, String requestId)
+            throws DataException {
+        CurrentPeriodEntity currentPeriodEntity = currentPeriodRepository.findById(id)
+                .orElseThrow(() -> new DataException(
+                        "Failed to add get Current period entity to add link"));
+        CurrentPeriodDataEntity currentPeriodDataEntity = currentPeriodEntity.getData();
+        Map<String, String> map = currentPeriodDataEntity.getLinks();
+        map.put(linkType.getLink(), link);
+        currentPeriodDataEntity.setLinks(map);
+        currentPeriodEntity.setData(currentPeriodDataEntity);
+        try {
+            currentPeriodRepository.save(currentPeriodEntity);
+        } catch (MongoException me) {
+            final Map<String, Object> debugMap = new HashMap<>();
+            debugMap.put("id", id);
+            debugMap.put("link", link);
+            debugMap.put("link_type", linkType.getLink());
+
+            DataException dataException = new DataException(
+                    "Failed to add link to Small full", me);
+            LOGGER.errorContext(requestId, dataException, debugMap);
+            throw dataException;
+        }
+    }
+
+    @Override
+    public String generateID(String value) {
+        String unencryptedId = value + "-" + ResourceName.CURRENT_PERIOD.getName();
+        byte[] id = messageDigest.digest(
```

Comment:
Rather than repeating this messageDigest/Base64 encoding logic could we extract to a helper/utility class method that just takes the unencryptedId as an argument?
This will make it easier to pull this functionality out into a library later on too so that we can use the same logic across services